.github/workflows/dev-setup.yml

```diff
@@ -0,0 +1,90 @@
+name: Dev environment setup
```

Comment:
Do we need to run this workflow in any PRs? or only when we change `dev/dev-env-setup.sh`?

Reply:
I think we'd likely be running this on a weekly schedule. I really don't want to run this on PRs or during nightly builds since it's going to impact our PR feedback heavily for something that really shouldn't be changing very often at all.

Reply:
I think we can do something similar to the cross-version test. Run this job weekly and when we update `dev/dev-env-setup.sh`.